qml/PingTextField.qml

```diff
@@ -5,46 +5,56 @@ import QtQuick.Layouts 1.3
 
 Item {
     id: root
-    property alias title: text.text
-    property var text: ""
+    property alias title: label.text
+    property string text: ""
     property alias validator: textField.validator
 
-    width: text.width + textField.width
-    height: text.height
+    width: label.width + textField.width
+    height: label.height
 
     // Emited when edition if finished with enter/return key or when TextField loses focus
-    signal editingFinished(var input)
+    signal editingFinished()
 
     // This avoid connection interactions while user input is on
     onTextChanged: {
         if(textField.focus) {
             return
         }
 
-        textField.text = root.text
+        textField.text = text
     }
 
 
     Label {
-        id: text
+        id: label
         color: Material.accent
     }
 
     TextField {
         id: textField
 
-        anchors.left: text.right
-        anchors.baseline: text.baseline
+        anchors.left: label.right
+        anchors.baseline: label.baseline
         anchors.leftMargin: 5
         selectByMouse: true
 
-        onEditingFinished: {
+        // editingFinished() is only emitted when TextField has focus
+        // That's why we are using accepted()
+        onAccepted: {
             focus = false
             mainPage.forceActiveFocus()
-            root.editingFinished(textField.text)
+            // It's necessary to update the input variable
+            root.text = textField.text
+            root.editingFinished()
         }
 
         onActiveFocusChanged: {
+            // TODO: We should update the user about wrong inputs here
+            // This allow us to correct not valid inputs to the last valid value
+            if(!acceptableInput) {
```

Comment:
`acceptableInput` is not defined, do you mean to comment this code block along with the comment?